server/src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -213,8 +213,8 @@ private boolean hasNewNodes(DiscoveryNodes nodes) {
 
     abstract class InternalAsyncFetch<T extends BaseNodeResponse> extends AsyncShardFetch<T> {
 
-        InternalAsyncFetch(Logger logger, String type, ShardId shardId, String customDataPath) {
-            super(logger, type, shardId, customDataPath);
+        InternalAsyncFetch(Logger logger, String type, ShardId shardId, String customDataPath, int nodeNumber) {
```

Comment:
naming nit/suggestion

```suggestion
        InternalAsyncFetch(Logger logger, String type, ShardId shardId, String customDataPath, int expectedSize) {
```